src/WebJobs.Script.WebHost/Swagger/SwaggerDocumentManager.cs

```diff
@@ -0,0 +1,204 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Net.Http;
+using System.Threading.Tasks;
+using System.Web.Http;
+using System.Web.Http.Routing;
+using Microsoft.Azure.WebJobs.Host;
+using Microsoft.Azure.WebJobs.Script.Config;
+using Microsoft.Azure.WebJobs.Script.WebHost.Models;
+using Microsoft.Azure.WebJobs.Script.WebHost.Properties;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Azure.WebJobs.Script.WebHost
+{
+    public class SwaggerDocumentManager : ISwaggerDocumentManager
+    {
+        private readonly TraceWriter _traceWriter;
+        private readonly string _swaggerFilePath;
+        private readonly string _swaggerDirPath;
+        private readonly ScriptHostConfiguration _config;
+
+        public SwaggerDocumentManager(ScriptHostConfiguration hostConfig)
+        {
+            _config = hostConfig;
+            _traceWriter = _config.TraceWriter;
+            _swaggerDirPath = Path.Combine(_config.RootScriptPath, ScriptConstants.ConfigDirName, ScriptConstants.SwaggerDirName);
+            _swaggerFilePath = Path.Combine(_swaggerDirPath, ScriptConstants.SwaggerFileName);
+        }
+
+        public async Task<JObject> GetSwaggerDocumentAsync()
+        {
+            if (_config.SwaggerDocumentMode == SwaggerDocumentMode.Manual)
+            {
+                return await ReadSwaggerAsync();
+            }
+            return null;
+        }
+
+        public JObject GenerateSwaggerDocument(HttpRouteCollection httpRoutes)
+        {
+            var swaggerDocument = new SwaggerDocument();
+            try
+            {
+                _traceWriter.Verbose(Resources.SwaggerGenerateDocument);
+                string hostname = ScriptSettingsManager.Instance.GetSetting(EnvironmentSettingNames.AzureWebsiteHostName);
+                swaggerDocument.SwaggerInfo.FunctionAppName = hostname;
+                swaggerDocument.Host = hostname;
+                swaggerDocument.ApiEndpoints = GetEndpointsData(httpRoutes);
+            }
+            catch (Exception ex)
```

Comment:
Wondering if we really want to swallow exceptions here. I think the caller should handle exceptions.